app/src/main/java/com/keylesspalace/tusky/pager/AccountPagerAdapter.kt

```diff
@@ -0,0 +1,66 @@
+/* Copyright 2019 Tusky Contributors
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.pager
+
+import android.util.SparseArray
+import androidx.fragment.app.*
+
+import com.keylesspalace.tusky.fragment.AccountMediaFragment
+import com.keylesspalace.tusky.fragment.TimelineFragment
+import com.keylesspalace.tusky.interfaces.RefreshableFragment
+
+import androidx.viewpager2.adapter.FragmentStateAdapter
+import java.lang.ref.WeakReference
+
+class AccountPagerAdapter(
+        activity: FragmentActivity,
+        private val accountId: String
+) : FragmentStateAdapter(activity) {
+
+    private val fragments = SparseArray<WeakReference<Fragment>>(TAB_COUNT)
```

Comment:
This is a hack I needed to use multiple times because there seems to be no out-of-the-box way to get the active fragments from a `FragmentStateAdapter`. Seems to work fine.

Reply:
Why `SparseArray` with only 4 tabs?